homeassistant/components/myuplink/sensor.py

```diff
@@ -56,17 +95,30 @@ async def async_setup_entry(
     # Setup device point sensors
     for device_id, point_data in coordinator.data.points.items():
         for point_id, device_point in point_data.items():
-            entities.append(
-                MyUplinkDevicePointSensor(
-                    coordinator=coordinator,
-                    device_id=device_id,
-                    device_point=device_point,
-                    entity_description=DEVICE_POINT_DESCRIPTIONS.get(
-                        device_point.parameter_unit
-                    ),
-                    unique_id_suffix=point_id,
+            description = get_description(device_point)
+            if (
+                description is not None
+                and description.device_class == SensorDeviceClass.ENUM
+            ):
+                entities.append(
+                    MyUplinkEnumSensor(
```

Comment:
We could store the entity class to use in a variable and move the entity instantiation out of the if/else block.

Reply:
Good idea!